src/core/control/ToolHandler.cpp

```diff
@@ -534,13 +534,18 @@ void ToolHandler::setSelectionEditTools(bool setColor, bool setSize, bool setFil
         t->setSize(TOOL_SIZE_NONE);
         t->setColor(Color(-1));
         t->setFill(false);
+
+        LineStyle newStyle = t->getLineStyle();
+        newStyle.setClearLineStyle(!setLineStyle);
+        t->setLineStyle(newStyle);
```

Comment:
```suggestion
        t->setCapability(TOOL_CAP_LINE_STYLE, setLineStyle);
        t->setSize(TOOL_SIZE_NONE);
        t->setColor(Color(-1));
        t->setFill(false);
```

with TOOL_CAP_LINE_STYLE declared in ToolEnums.h

Reply:
Resolved by baffc58cbe83b76317ee8a0c8f5497dd5b52c85b.